tssc/step_implementers/validate_environment_configuration/configlint.py

```diff
@@ -0,0 +1,231 @@
+| `rules`           | False    | ./config_lint.rules       | File containing user-defined rules
+
+Expected Previous Step Results
+------------------------------
+
+Results expected from previous steps:
+
+| Step Name           |  Key                | Description
+|---------------------|---------------------|------------
+| `validate`          | `options.yml_path`  | Provides the path to the yml files
+| `environment-`      |                     | to be evaluated
+| `configuration`     |                     |
+
+Results
+-------
+
+Results output by this step:
+
+| Key                              | Description
+|----------------------------------|------------
+| `result`                         | A dictionary describing the output \
+                                     of this step
+| `report-artifacts`               | An array of dictionaries describing \
+                                     artifacts generated by this step
+
+Elements in `result` dictionary:
+
+| Key          | Description
+|--------------|------------
+| `success`    | Boolean value describing success/failure of this step
+| `message`    | Human readable message describing results of this step
+
+Elements in `report-artifacts` dictionary:
+
+|  Key         | Description
+|--------------|------------
+| `name`       | Human readable name for report artifact generated by this step
+| `path`       | Absolute path (including transport protocol) to the step report artifact
+
+
+Examples
+--------
+
+**Example: Step Configuration (minimal)**
+
+    validate-environment-configuration:
+    - implementer: ConfiglintFromArgocd
+    - implementer: Configlint
+      config:
+        rules: 'config_lint.rules'
+
+.. Note:: The configuration example above uses the ConfiglintFromArgocd step to \
+provide the yml_path option.
+
+**Example: Generated Config Lint Call (runtime)**
+
+    config-lint -verbose
+        -rules rules.yml /home/user/tssc-working/deploy/file.yml
+
+**Example: Existing Rules File (minimal)**
+
+    version: 1
+    description: Rules for Kubernetes spec files
+    type: Kubernetes
+    files:
+      - "*.yml"
+    rules:
+    - id: TSSC_EXAMPLE
+      severity: FAILURE
+      message: Deployment must have istio
+      resource: Deployment
+      assertions:
+        - key: spec.template.metadata.annotations
+          op: contains
+          value: '"sidecar.istio.io/inject": false'
+
+Example: Results
+
+    'tssc-results': {
+        'validate-environment-configuration': {
+            'result': {
+                'success': True,
+                'message': 'config-lint step completed'
+            }
+            'options': {
+                'yml_path': '/home/user/tssc-working/file.yml'
+            }
+
+    }
+
+.. Note:: The configuration example above used the ConfiglintFromArgocd step to \
+provide the yml_path option.
+
+"""
+
+import os
+import sys
+import sh
+from tssc import StepImplementer
+
+DEFAULT_CONFIG = {
+    'rules': './config-lint.rules'
```

Comment:
Are rules always contained in a single file? Looking at [this example set](https://github.com/stelligent/config-lint/blob/master/example-files/rules), I could image situations where it's checked against multiple rule-sets. Perhaps out of scope for this effort..

Reply:
+1, maybe a backlog jira

Reply:
This functionality exists.  
It is a matter of changing the tssc-config.yml to use a folder instead of a file; the reference application has been updated with TWO rules files to set the pattern.